docs/troubleshooting.md

```diff
@@ -0,0 +1,249 @@
+
+```sh
+$ cat my-subscription.yaml
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: my-subscription
+  namespace: my-subscription-namespace
+spec:
+  channel: beta
+  name: prometheus
+  source: my-catalogsource
+  sourceNamespace: my-catalogsource-namespace
+  installPlanApproval: Automatic
+```
+
+After creating the `Subscription` you should check its status to confirm that it has been successfully installed. The Catalog operator will constantly update the Status of a `Subscription` that it manages with useful information regarding its current state. You can check the status of your `Subscription` with the following command:
+
+```sh
+$ kubectl -n my-subscription-namespace get subscriptions my-subscription -o yaml | yq r - status
+conditions:
+- lastTransitionTime: "2019-11-20T21:00:17Z"
+  message: dependency resolution requires at least one catalogsource
+  reason: NoCatalogSourcesFound
+  status: "True"
+  type: CatalogSourcesUnhealthy
+lastUpdated: "2019-11-20T21:00:17Z"
+```
+
+>Note: It is possible that the Status is missing, which suggests that the Catalog operator is encountering an issue when processing the CSV in a very early stage. You should respond by reviewing the logs of the Catalog operator.
+
+This is odd - we had already confirmed in the [previous scenario](#catalogsource-troubleshooting) that the `CatalogSource` exists. As such, we should review the Catalog operator logs:
+
+```sh
+$ kubectl -n olm get pods
+NAME                                READY   STATUS    RESTARTS   AGE
+catalog-operator-5bdc79c56b-htn29   1/1     Running   0          4m39s
+olm-operator-6999db5767-zk4qp       1/1     Running   0          4m40s
+packageserver-6f8547dcb9-7wvtj      1/1     Running   0          4m36s
+
+$ kubectl -n olm logs catalog-operator-5bdc79c56b-htn29
+time="2019-11-21T14:40:21Z" level=debug msg="resolving subscriptions in namespace" id=m1OUy namespace=my-subscription-namespace
+E1121 14:40:21.066187       1 queueinformer_operator.go:282] sync "my-subscription-namespace" failed: no catalog sources available
+time="2019-11-21T14:40:21Z" level=debug msg="resolving sources" id=eFa1R namespace=my-subscription-namespace
+time="2019-11-21T14:40:21Z" level=debug msg="checking if subscriptions need update" id=eFa1R namespace=my-subscription-namespace
+time="2019-11-21T14:40:21Z" level=debug msg="checking for existing installplan" channel=beta id=eFa1R namespace=my-subscription-namespace pkg=prometheus source=my-catalogsource sub=my-subscription
+```
+
+From the logs it's clear that the Catalog operator believes that there are no `CatalogSources` available. This is due to the fact that `Subscriptions` cannot install operators provided by `CatalogSources` that are not in the same namespace unless the `CatalogSource` is created in the `olm namespace`. We can fix this issue by moving the `Subscription` to the `my-catalogsource-namespace namespace`.
```

Comment:
Do we want to first tell the reader to create the `Subscription` in the same namespace, instead of doing the wrong this and telling the reader, "hah, gotcha" :p ?  I feel like a lot of the readers will tend to follow along the doc, instead of reading the entire doc once and then coming back to do the right thing.

Reply:
Hey @anik120 - that's a fair point. I designed this to highlight any potential "gotcha" that an OLM user may encounter. There are other docs that are meant to teach users how they should approach use OLM.

Perhaps it would be good to highlight at the top of the dock that "Everything  that can go wrong, will go wrong in this doc."

Reply:
Doc has been revamped.